src/pages/workspace/accounting/netsuite/import/NetSuiteImportCustomFieldNew/NetSuiteImportAddCustomListPage.tsx

```diff
@@ -0,0 +1,176 @@
+import InteractiveStepSubHeader from '@components/InteractiveStepSubHeader';
+import type {InteractiveStepSubHeaderHandle} from '@components/InteractiveStepSubHeader';
+import useLocalize from '@hooks/useLocalize';
+import useSubStep from '@hooks/useSubStep';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as Connections from '@libs/actions/connections/NetSuiteCommands';
+import Navigation from '@libs/Navigation/Navigation';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import type {CustomFieldSubStepWithPolicy} from '@pages/workspace/accounting/netsuite/types';
+import type {WithPolicyConnectionsProps} from '@pages/workspace/withPolicyConnections';
+import withPolicyConnections from '@pages/workspace/withPolicyConnections';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import INPUT_IDS from '@src/types/form/NetSuiteCustomFieldForm';
+import ChooseCustomListStep from './substeps/ChooseCustomListStep';
+import ConfirmCustomListStep from './substeps/ConfirmCustomListStep';
+import MappingStep from './substeps/MappingStep';
+import TransactionFieldIDStep from './substeps/TransactionFieldIDStep';
+
+const formSteps = [ChooseCustomListStep, TransactionFieldIDStep, MappingStep, ConfirmCustomListStep];
+
+function NetSuiteImportAddCustomListPage({policy}: WithPolicyConnectionsProps) {
+    const policyID = policy?.id ?? '-1';
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const ref: ForwardedRef<InteractiveStepSubHeaderHandle> = useRef(null);
+
+    const config = policy?.connections?.netsuite?.options?.config;
+    const customLists = useMemo(() => config?.syncOptions?.customLists ?? [], [config?.syncOptions]);
+
+    const handleFinishStep = useCallback(() => {
+        Navigation.goBack(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_FIELD_MAPPING.getRoute(policyID, CONST.NETSUITE_CONFIG.IMPORT_CUSTOM_FIELDS.CUSTOM_LISTS));
+    }, [policyID]);
+
+    const {
+        componentToRender: SubStep,
+        isEditing,
+        nextScreen,
+        prevScreen,
+        screenIndex,
+        moveTo,
+        goToTheLastStep,
+    } = useSubStep<CustomFieldSubStepWithPolicy>({
+        bodyContent: formSteps,
+        startFrom: CONST.NETSUITE_CUSTOM_FIELD_SUBSTEP_INDEXES.CUSTOM_LISTS.CUSTOM_LIST_PICKER,
+        onFinished: handleFinishStep,
+    });
+
+    const handleBackButtonPress = () => {
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+
+        // Clicking back on the first screen should go back to listing
+        if (screenIndex === CONST.NETSUITE_CUSTOM_FIELD_SUBSTEP_INDEXES.CUSTOM_LISTS.CUSTOM_LIST_PICKER) {
+            Navigation.goBack(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_FIELD_MAPPING.getRoute(policyID, CONST.NETSUITE_CONFIG.IMPORT_CUSTOM_FIELDS.CUSTOM_LISTS));
+            return;
+        }
+        ref.current?.movePrevious();
+        prevScreen();
+    };
```

Comment:
We are not clearing the form errors on going back which is causing issue https://github.com/Expensify/App/issues/49266, we do need to clear errors because we are using the same Form by just changing the content of the sub-step.